sw/device/tests/dif/dif_otbn_sanitytest.c

```diff
@@ -0,0 +1,252 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_otbn.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+/**
+ * Information about an embedded OTBN application image.
+ *
+ * All pointers reference data in the normal CPU address space.
+ */
+typedef struct {
+  const char *imem_start;  //!< start of OTBN instruction memory
+  const char *imem_end;    //!< end of OTBN instruction memory
+  const char *dmem_start;  //!< start of OTBN data memory
+  const char *dmem_end;    //!< end of OTBN data memory
+} otbn_app_t;
+
+/**
+ * A function (entry point) in an OTBN application.
+ */
+typedef struct {
+  const otbn_app_t *app;  //!< application associated with this function
+  const char *func;       //!< address of the function entry point in normal CPU
+                          //!< address space
+} otbn_func_t;
+
+// Prepare info structs representing the barrett384 application, and the
+// wrap_barrett384() function. The goal of these structs is to make the linker-
+// generated symbols for the application image available in a form that is
+// easier to use.
+//
+// TODO: Think about macro-ifying or auto-generating some of this code in a
+// future OTBN driver.
+
+// See https://sourceware.org/binutils/docs/ld/Source-Code-Reference.html for
+// an in-depth explanation of how symbols are made available to C code.
+extern const char _otbn_app_barrett384_imem_start[];
+extern const char _otbn_app_barrett384_imem_end[];
+extern const char _otbn_app_barrett384_dmem_start[];
+extern const char _otbn_app_barrett384_dmem_end[];
+extern const char _otbn_app_barrett384_wrap_barrett384[];
+
+const otbn_app_t kOtbnAppBarrett384 = {
+    .imem_start = _otbn_app_barrett384_imem_start,
+    .imem_end = _otbn_app_barrett384_imem_end,
+    .dmem_start = _otbn_app_barrett384_dmem_start,
+    .dmem_end = _otbn_app_barrett384_dmem_end,
+};
+
+const otbn_func_t kOtbnFuncBarrett384WrapBarrett384 = {
+    .app = &kOtbnAppBarrett384,
+    .func = _otbn_app_barrett384_wrap_barrett384,
+};
+
+const test_config_t kTestConfig = {
+    .can_clobber_uart = false,
+};
+
+static dif_otbn_t otbn;
+
+/*
+// Dump OTBN's memory. A development helper, unused otherwise.
+static void otbn_dump_dmem() {
+  uint32_t data[8];
+  for (int i = 0; i < dif_otbn_get_dmem_size_bytes(&otbn) / 32; ++i) {
+    dif_otbn_dmem_read(&otbn, i * 32, data, 32);
+    LOG_INFO("DMEM @%04d: 0x%08x%08x%08x%08x%08x%08x%08x%08x", i, data[7],
+             data[6], data[5], data[4], data[3], data[2], data[1], data[0]);
+  }
+}
+*/
```

Comment:
Did this get left by mistake...?

Reply:
No, it's in there intentionally as development helper when bringing up and debugging OTBN applications.

Reply:
Why not uncomment it, then? I don't think we should be checking in commented-out code.

Reply:
Honestly, this shouldn't hold up merging this test.